builder/vagrant/step_initialize_vagrant.go

```diff
@@ -0,0 +1,136 @@
+package vagrant
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type StepInitializeVagrant struct {
+	BoxName      string
+	BoxVersion   string
+	Minimal      bool
+	Template     string
+	SourceBox    string
+	OutputDir    string
+	SyncedFolder string
+	GlobalID     string
+}
+
+var DEFAULT_TEMPLATE = `Vagrant.configure("2") do |config|
+  config.vm.box = "{{.BoxName}}"
+  {{ if ne .SyncedFolder "" -}}
+  		config.vm.synced_folder "{{.SyncedFolder}}", "/vagrant"
+  {{- else -}}
+  		config.vm.synced_folder ".", "/vagrant", disabled: true
+  {{- end}}
+end`
+
+type VagrantfileOptions struct {
+	SyncedFolder string
+	BoxName      string
+}
+
+func (s *StepInitializeVagrant) createInitializeCommand() (string, error) {
+	tplPath := filepath.Join(s.OutputDir, "packer-vagrantfile-template.erb")
+	templateFile, err := os.Create(tplPath)
+	templateFile.Chmod(0777)
```

Comment:
I don't think that `packer-vagrantfile-template.erb` needs exec (7) rights as Vagrant will just read it. `os.Create` will give (rw, 6, 0666), which should be enough I think.
But if I am wrong, look inside `os.Create`, it calls: `os.OpenFile(name, O_RDWR|O_CREATE|O_TRUNC, 0666)`, these two calls can be merged together.

Reply:
Lol nope you're right, this was a debug line that isn't necessary.